homeassistant/components/bmw_connected_drive/select.py

```diff
@@ -0,0 +1,140 @@
+"""Select platform for BMW."""
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from bimmer_connected.vehicle import MyBMWVehicle
+from bimmer_connected.vehicle.charging_profile import ChargingMode
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfElectricCurrent
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BMWBaseEntity
+from .const import DOMAIN
+from .coordinator import BMWDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class BMWRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    current_option: Callable[[MyBMWVehicle], Any]
+    remote_service: Callable[[MyBMWVehicle, Any], Any]
```

Comment:
```suggestion
    remote_service: Callable[[MyBMWVehicle, str], Coroutine[Any, Any, Any]]
```
Or you could also use `Callable[[RemoteServices, str], Coroutine[Any, Any, Any]]`